packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1451,12 +1449,12 @@ class _BackLabel extends StatelessWidget {
   Widget build(BuildContext context) {
     if (specifiedPreviousTitle != null) {
       return _buildPreviousTitleWidget(context, specifiedPreviousTitle, null);
-    } else if (route is CupertinoRouteTransitionMixin<dynamic> && !route.isFirst) {
+    } else if (route is CupertinoRouteTransitionMixin<dynamic> && !route!.isFirst) {
```

Comment:
Shouldn't the compiler assume that `route` is non-nullable after the `is` comparing it to a non-nullable type?

Reply:
promotion to non-null does not work for fields/getters. See dart-lang/language#1187 for possible improvement.

Reply:
Darn.  OK.